src/duckling/canvas/tools/entity-move-tool.ts

```diff
@@ -61,6 +64,24 @@ export class EntityMoveTool extends BaseTool {
         this._cancel();
     }
 
+    onKeyDown(event : CanvasKeyEvent) {
+        if (this._selectionService.selection.value.entity) {
+            let oldPosition = getPosition(this._selectionService.selection.value.entity).position;
+            let adjustment = this._keyEventToPositionAdjustment(event.keyCode);
+            let newPosition = {
```

Comment:
You should consider defining a library of vector routines. Example: vectorAdd and vectorRound. That could clean up several of the operations in this file.